tools/draincluster/drain/drain.go

```diff
@@ -0,0 +1,244 @@
+				ObjectMeta: metav1.ObjectMeta{
+					Name: evictionName,
+				},
+				Spec: placementv1beta1.PlacementEvictionSpec{
+					PlacementName: crpName,
+					ClusterName:   h.ClusterName,
+				},
+			}
+			return h.HubClient.Create(ctx, &eviction)
+		})
+
+		if err != nil {
+			return false, fmt.Errorf("failed to create eviction for CRP %s: %w", crpName, err)
+		}
+	}
+
+	// wait until all evictions reach a terminal state.
+	for crpName := range h.ClusterResourcePlacementResourcesMap {
+		err := wait.ExponentialBackoffWithContext(ctx, retry.DefaultBackoff, func(ctx context.Context) (bool, error) {
+			evictionName := fmt.Sprintf(drainEvictionNameFormat, crpName, h.ClusterName)
+			eviction := placementv1beta1.ClusterResourcePlacementEviction{}
+			if err := h.HubClient.Get(ctx, types.NamespacedName{Name: evictionName}, &eviction); err != nil {
+				return false, fmt.Errorf("failed to get eviction %s: %w", evictionName, err)
+			}
+			return evictionutils.IsEvictionInTerminalState(&eviction), nil
+		})
+
+		if err != nil {
+			return false, fmt.Errorf("failed to wait for evictions to reach terminal state: %w", err)
+		}
+	}
+
+	isDrainSuccessful := true
+	for crpName := range h.ClusterResourcePlacementResourcesMap {
+		evictionName := fmt.Sprintf(drainEvictionNameFormat, crpName, h.ClusterName)
+		eviction := placementv1beta1.ClusterResourcePlacementEviction{}
+		if err := h.HubClient.Get(ctx, types.NamespacedName{Name: evictionName}, &eviction); err != nil {
+			return false, fmt.Errorf("failed to get eviction %s: %w", evictionName, err)
+		}
+		validCondition := eviction.GetCondition(string(placementv1beta1.PlacementEvictionConditionTypeValid))
+		if validCondition != nil && validCondition.Status == metav1.ConditionFalse {
+			// check to see if CRP is missing or CRP is being deleted or CRB is missing.
+			if validCondition.Reason == condition.EvictionInvalidMissingCRPMessage ||
+				validCondition.Reason == condition.EvictionInvalidDeletingCRPMessage ||
+				validCondition.Reason == condition.EvictionInvalidMissingCRBMessage {
+				continue
+			}
+		}
+		executedCondition := eviction.GetCondition(string(placementv1beta1.PlacementEvictionConditionTypeExecuted))
+		if executedCondition == nil || executedCondition.Status == metav1.ConditionFalse {
+			isDrainSuccessful = false
+			log.Printf("eviction %s was not executed successfully for CRP %s", evictionName, crpName)
+			continue
+		}
+		log.Printf("eviction %s was executed successfully for CRP %s", evictionName, crpName)
+		// log each resource evicted by CRP.
+		for i := range h.ClusterResourcePlacementResourcesMap[crpName] {
+			resourceIdentifier := h.ClusterResourcePlacementResourcesMap[crpName][i]
+			log.Printf("evicted resource %s propagated by CRP %s", fmt.Sprintf(resourceIdentifierKeyFormat, resourceIdentifier.Group, resourceIdentifier.Version, resourceIdentifier.Kind, resourceIdentifier.Name, resourceIdentifier.Namespace), crpName)
```

Comment:
Hi Arvind! The format param order seems to be wrong? It should be GVK -> namespace -> name.